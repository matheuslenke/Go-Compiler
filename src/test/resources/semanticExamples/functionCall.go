package funcoes

import "fmt"

func funcao1() string {
	var str = "Hello World"

	return str
}

func somaDoisNumeros(a, b int) int {
	var c int = a + b

	return c
}

func retornaDobro(a float32) float32 {
	var c = a * 2.0
	return c
}

func funcao2() {
	funcao1()
	var a int = 1
	var b int = 2
	var c float32 = 5.5
	var result int
	var array [2]int
	array[1] = somaDoisNumeros(a, b)

	result = somaDoisNumeros(array[0], array[1])

	c = retornaDobro(c)
}
