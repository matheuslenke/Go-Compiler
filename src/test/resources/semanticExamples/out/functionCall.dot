digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) funcao1@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="result"];
node13[label="(string) type_use"];
node12 -> node13;
node10 -> node12;
node14[label="block"];
node15[label="var_list"];
node16[label="="];
node17[label="(string) str@6"];
node16 -> node17;
node18[label="(string) string@0"];
node16 -> node18;
node15 -> node16;
node14 -> node15;
node19[label="return"];
node20[label="(string) str@6"];
node19 -> node20;
node14 -> node19;
node10 -> node14;
node0 -> node10;
node21[label="(func) somaDoisNumeros@7"];
node22[label="parameters"];
node23[label="(int) a@8"];
node22 -> node23;
node24[label="(int) b@9"];
node22 -> node24;
node21 -> node22;
node25[label="result"];
node26[label="(int) type_use"];
node25 -> node26;
node21 -> node25;
node27[label="block"];
node28[label="var_list"];
node29[label="="];
node30[label="(int) c@10"];
node29 -> node30;
node31[label="(int) +"];
node32[label="(int) a@8"];
node31 -> node32;
node33[label="(int) b@9"];
node31 -> node33;
node29 -> node31;
node28 -> node29;
node27 -> node28;
node34[label="return"];
node35[label="(int) c@10"];
node34 -> node35;
node27 -> node34;
node21 -> node27;
node0 -> node21;
node36[label="(func) retornaDobro@11"];
node37[label="parameters"];
node38[label="(float) a@12"];
node37 -> node38;
node36 -> node37;
node39[label="result"];
node40[label="(float) type_use"];
node39 -> node40;
node36 -> node39;
node41[label="block"];
node42[label="var_list"];
node43[label="="];
node44[label="(float) c@13"];
node43 -> node44;
node45[label="(float) *"];
node46[label="(float) a@12"];
node45 -> node46;
node47[label="(float) real@2.0"];
node45 -> node47;
node43 -> node45;
node42 -> node43;
node41 -> node42;
node48[label="return"];
node49[label="(float) c@13"];
node48 -> node49;
node41 -> node48;
node36 -> node41;
node0 -> node36;
node50[label="(func) funcao2@14"];
node51[label="parameters"];
node50 -> node51;
node52[label="block"];
node53[label="func_use"];
node54[label="(func) funcao1@5"];
node53 -> node54;
node55[label="arguments"];
node53 -> node55;
node52 -> node53;
node56[label="var_list"];
node57[label="="];
node58[label="(int) a@15"];
node57 -> node58;
node59[label="(int) int@1"];
node57 -> node59;
node56 -> node57;
node52 -> node56;
node60[label="var_list"];
node61[label="="];
node62[label="(int) b@16"];
node61 -> node62;
node63[label="(int) int@2"];
node61 -> node63;
node60 -> node61;
node52 -> node60;
node64[label="var_list"];
node65[label="="];
node66[label="(float) c@17"];
node65 -> node66;
node67[label="(float) real@5.5"];
node65 -> node67;
node64 -> node65;
node52 -> node64;
node68[label="var_list"];
node69[label="(int) result@18"];
node68 -> node69;
node52 -> node68;
node70[label="var_list"];
node71[label="(array) array@19"];
node72[label="array_length"];
node73[label="(int) int@2"];
node72 -> node73;
node71 -> node72;
node74[label="(int) element_type"];
node71 -> node74;
node70 -> node71;
node52 -> node70;
node75[label="assign_list_node"];
node76[label="="];
node77[label="(array) array@19"];
node78[label="(int) index"];
node79[label="(int) int@1"];
node78 -> node79;
node77 -> node78;
node76 -> node77;
node80[label="func_use"];
node81[label="(func) somaDoisNumeros@7"];
node80 -> node81;
node82[label="arguments"];
node83[label="(int) a@15"];
node82 -> node83;
node84[label="(int) b@16"];
node82 -> node84;
node80 -> node82;
node76 -> node80;
node75 -> node76;
node52 -> node75;
node85[label="assign_list_node"];
node86[label="="];
node87[label="(int) result@18"];
node86 -> node87;
node88[label="func_use"];
node89[label="(func) somaDoisNumeros@7"];
node88 -> node89;
node90[label="arguments"];
node91[label="(array) array@19"];
node92[label="(int) index"];
node93[label="(int) int@0"];
node92 -> node93;
node91 -> node92;
node90 -> node91;
node94[label="(array) array@19"];
node95[label="(int) index"];
node96[label="(int) int@1"];
node95 -> node96;
node94 -> node95;
node90 -> node94;
node88 -> node90;
node86 -> node88;
node85 -> node86;
node52 -> node85;
node97[label="assign_list_node"];
node98[label="="];
node99[label="(float) c@17"];
node98 -> node99;
node100[label="func_use"];
node101[label="(func) retornaDobro@11"];
node100 -> node101;
node102[label="arguments"];
node103[label="(float) c@17"];
node102 -> node103;
node100 -> node102;
node98 -> node100;
node97 -> node98;
node52 -> node97;
node50 -> node52;
node0 -> node50;
}
