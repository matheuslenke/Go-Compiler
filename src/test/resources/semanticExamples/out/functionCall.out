[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: Hello World




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: funcao1, line: 5, type: func, scopeName: global
Entry 6 -- name: str, line: 6, type: string, scopeName: funcao1
Entry 7 -- name: somaDoisNumeros, line: 11, type: func, scopeName: global
Entry 8 -- name: a, line: 11, type: int, scopeName: somaDoisNumeros
Entry 9 -- name: b, line: 11, type: int, scopeName: somaDoisNumeros
Entry 10 -- name: c, line: 12, type: int, scopeName: somaDoisNumeros
Entry 11 -- name: retornaDobro, line: 17, type: func, scopeName: global
Entry 12 -- name: a, line: 17, type: float, scopeName: retornaDobro
Entry 13 -- name: c, line: 18, type: float, scopeName: retornaDobro
Entry 14 -- name: funcao2, line: 22, type: func, scopeName: global
Entry 15 -- name: a, line: 24, type: int, scopeName: funcao2
Entry 16 -- name: b, line: 25, type: int, scopeName: funcao2
Entry 17 -- name: c, line: 26, type: float, scopeName: funcao2
Entry 18 -- name: result, line: 27, type: int, scopeName: funcao2
Entry 19 -- name: array, line: 28, type: array, subtype: int, scopeName: funcao2




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.866 s
[INFO] Finished at: 2022-02-24T20:53:46-03:00
[INFO] ------------------------------------------------------------------------
